app_dart/lib/src/request_handlers/reset_prod_task.dart

```diff
@@ -41,39 +41,54 @@ class ResetProdTask extends ApiRequestHandler<Body> {
 
   @override
   Future<Body> post() async {
-    checkRequiredParameters(<String>[taskKeyParam]);
     final DatastoreService datastore = datastoreProvider(config.db);
-    final String encodedKey = requestData[taskKeyParam] as String;
+    final String encodedKey = requestData[taskKeyParam] as String ?? '';
     final ClientContext clientContext = authContext.clientContext;
     final KeyHelper keyHelper = KeyHelper(applicationContext: clientContext.applicationContext);
     final String owner = requestData[ownerParam] as String ?? 'flutter';
-    final String repo = requestData[repoParam] as String ?? '';
-    final Key<int> key = keyHelper.decode(encodedKey) as Key<int>;
-    log.info('Rescheduling task with Key: ${key.id}');
-    final Task task = (await datastore.lookupByKey<Task>(<Key<int>>[key])).single;
-    if (task == null) {
-      throw BadRequestException('No such task: $key');
-    }
-    // Task is complete and it succeeded, nothing to do here.
-    if (task.status == 'Succeeded') {
-      return Body.empty;
-    }
-    final Commit commit = await datastore.db.lookupValue<Commit>(task.commitKey, orElse: () {
-      throw BadRequestException('No such commit: ${task.commitKey}');
-    });
-    String builder = task.builderName;
-    if (builder == null) {
-      final List<LuciBuilder> builders = await config.luciBuilders('prod', commit.slug);
-      builder = builders
-          .where((LuciBuilder builder) => builder.taskName == task.name)
-          .map((LuciBuilder builder) => builder.name)
-          .single;
+    final String repo = requestData[repoParam] as String ?? 'flutter';
+    String commitSha = requestData[commitShaParam] as String ?? '';
+    RepositorySlug slug;
+    String builder = requestData[builderParam] as String ?? '';
+    Task task;
+
+    if (encodedKey.isNotEmpty) {
+      // Request coming from the dashboard.
+      final Key<int> key = keyHelper.decode(encodedKey) as Key<int>;
+      log.info('Rescheduling task with Key: ${key.id}');
+      task = (await datastore.lookupByKey<Task>(<Key<int>>[key])).single;
+      if (task.status == 'Succeeded') {
+        return Body.empty;
+      }
+      final Commit commit = await datastore.db.lookupValue<Commit>(task.commitKey, orElse: () {
+        throw BadRequestException('No such commit: ${task.commitKey}');
+      });
+      slug = commit.slug;
+      commitSha = commit.sha;
+      builder = task.builderName;
+      if (builder == null) {
+        final List<LuciBuilder> builders = await config.luciBuilders('prod', slug);
+        builder = builders
+            .where((LuciBuilder builder) => builder.taskName == task.name)
+            .map((LuciBuilder builder) => builder.name)
+            .single;
+      }
+    } else {
+      // Request not coming from dashboard means we need to create everything from
+      // parameters.
+      if (commitSha.isEmpty || builder.isEmpty || repo.isEmpty) {
+        throw const BadRequestException(
```

Comment:
`checkRequiredParameters` makes this easier to read (see https://cs.opensource.google/flutter/cocoon/+/master:app_dart/lib/src/request_handlers/update_task_status.dart)

Reply:
Done